test/test_array_view.cpp

```diff
@@ -0,0 +1,65 @@
+/* Copyright (C) 2021, Nikolai Wuttke. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <base/array_view.hpp>
+#include <base/warnings.hpp>
+
+RIGEL_DISABLE_WARNINGS
+#include <catch.hpp>
+RIGEL_RESTORE_WARNINGS
+
+#include <array>
+#include <vector>
+
+TEST_CASE("Array view constexpr")
+{
+  SECTION("pointer and size")
+  {
+    static constexpr auto expectedSize = 5;
```

Comment:
Do these need to be `static`, or would `constexpr` be sufficient?

Reply:
well, the expected size is ok being constexpr, the things requiring addresses-  like the arrays - need to also have a storage specifier. from what I understand, seems like the compiler needs to put them somewhere to actually do the computation